### PR TITLE
Minor updates

### LINKED ISSUE #111: SUGGESTION: Catchall state for errors
I've run into the case where I want to query for jobs that didn't finish. I don't care about the state they got to, I just want to find those that failed. Currently, I can do 2 separate queries:

```
jf job list --state FAILED
jf job list --state REMOTE_ERROR
```

But it might be nice to have a way to do this in one query.

### LINKED ISSUE #118: Flow state stays COMPLETED when a job is rerun
When a Flow has completed, its state is COMPLETED. In some cases one wants to rerun a given job (e.g. a FAILED job for which the Flow went on because it allows failed parents). In that case the Flow state stays as COMPLETED.

### LINKED ISSUE #119: Allow setting resources for FAILED and COMPLETED jobs
Related to #118 , currently one needs to stop the runner, set the resources, rerun the job then restart the runner because you can't set resources on a FAILED or COMPLETED job. Allowing setting resources for FAILED and COMPLETED jobs (maybe with a "-force" ? or should it be always allowed ?)

### PR DESCRIPTION
**Pull request by gpetretto**
* update docs
* allow to query in CLI and JobController by multiple Job and Flow states
* `--error` option added to filter all types of failed jobs in `jf job list`
* allow to set Job resources/worker/exec_config also for Jobs in failed and completed states
* fix runner bug
* fix graph

### CODE REVIEW

File: src/jobflow_remote/jobs/jobcontroller.py [hunk 1]
```diff
@@ -2400,7 +2413,7 @@ def count_jobs(
         job_ids: tuple[str, int] | list[tuple[str, int]] | None = None,
         db_ids: str | list[str] | None = None,
         flow_ids: str | list[str] | None = None,
-        state: JobState | None = None,
+        states: JobState | None = None,
```
Comment (davidwaroquiers) [inline src/jobflow_remote/jobs/jobcontroller.py]:
```suggestion
        states: JobState | list[JobState] | None = None,
```

File: src/jobflow_remote/jobs/jobcontroller.py [hunk 2]
```diff
@@ -1809,7 +1813,7 @@ def set_job_run_properties(
...
```
Comment (davidwaroquiers) [inline src/jobflow_remote/jobs/jobcontroller.py]:
```suggestion
        states: JobState | list[JobState] | None = None,
```

File: src/jobflow_remote/jobs/jobcontroller.py [hunk 3]
```diff
@@ -2134,7 +2147,7 @@ def unlock_jobs(
...
```
Comment (davidwaroquiers) [inline src/jobflow_remote/jobs/jobcontroller.py]:
```suggestion
        states: JobState | list[JobState] | None = None,
```

File: tests/integration/test_slurm.py [hunk 4]
```diff
@@ -299,5 +299,5 @@ def test_undefined_additional_stores(worker, job_controller):
     assert job_controller.count_jobs({}) == 2
 
     # The job should fail, as the additional store is not defined
-    assert job_controller.count_jobs(state=JobState.COMPLETED) == 1
-    assert job_controller.count_jobs(state=JobState.REMOTE_ERROR) == 1
+    assert job_controller.count_jobs(states=JobState.COMPLETED) == 1
+    assert job_controller.count_jobs(states=JobState.REMOTE_ERROR) == 1
```
Comment (davidwaroquiers) [inline tests/integration/test_slurm.py]:
```suggestion
    assert job_controller.count_jobs(states=JobState.REMOTE_ERROR) == 1
    assert job_controller.count_jobs(states=[JobState.COMPLETED, JobState.REMOTE_ERROR]) == 2
```

File: src/jobflow_remote/cli/job.py [hunk 5]
```diff
@@ -98,6 +107,10 @@ def jobs_list(
     start_date = get_start_date(start_date, days, hours)
 
     db_sort: list[tuple[str, int]] = [(sort.value, 1 if reverse_sort else -1)]
+    print(error)
```
Comment (davidwaroquiers) [inline src/jobflow_remote/cli/job.py]: Should this be removed ?

File: src/jobflow_remote/jobs/graph.py [hunk 6]
```diff
@@ -25,6 +25,8 @@ def get_graph(flow: FlowInfo, label: str = "name") -> DiGraph:
         job_prop["job_name"] = job_prop.pop("name")
         graph.add_node(db_id, **job_prop)
 
+    print(flow.parents)
```
Comment (davidwaroquiers) [inline src/jobflow_remote/jobs/graph.py]: Should this be removed ?

**Review by davidwaroquiers — APPROVED** [review]
All good to me! Just a few prints to be removed, some type hints missing for the states arguments.

**Comment by gpetretto** [comment]
Thanks for the comments, I included the suggestions. 
Also added the fix for issue #118. @davidwaroquiers not sure if you want to review this again.

**Comment by davidwaroquiers** [comment]
> Thanks for the comments, I included the suggestions. Also added the fix for issue #118. @davidwaroquiers not sure if you want to review this again.

No fine with me you can merge.

**Pull request merged**